daemon/src/model/cfd.rs

```diff
@@ -1688,13 +1757,35 @@ pub struct RevokedCommit {
 pub struct TimestampedTransaction {
     pub tx: Transaction,
     pub timestamp: SystemTime,
+    #[serde(with = "::bdk::bitcoin::util::amount::serde::as_sat")]
+    payout: Amount,
 }
 
 impl TimestampedTransaction {
-    pub fn new(tx: Transaction) -> Self {
+    pub fn new(tx: Transaction, own_script_pubkey: Script) -> Self {
+        // Falls back to Amount::ZERO in case we don't find an output that matches out script pubkey
+        // The assumption is, that this can happen for cases where we were liuqidated
+        let payout = tx
+            .output
+            .iter()
+            .find(|output| output.script_pubkey == own_script_pubkey)
+            .map(|output| Amount::from_sat(output.value))
+            .unwrap_or_default();
+
+        if payout == Amount::ZERO {
+            tracing::error!(
+                "Collaborative settlement with a zero amount, this should really not happen!"
+            );
+        }
```

Comment:
In this case, I think matching on the `Option` returned from `find` and logging as part of returning `Amount::ZERO` is easier to understand:

```suggestion
        let payout = match tx
            .output
            .iter()
            .find(|output| output.script_pubkey == own_script_pubkey)
            .map(|output| Amount::from_sat(output.value)) {
      	    Some(payout) => payout,
      	    None => {
      	  	tracing::error!(
	            "Collaborative settlement with a zero amount, this should really not happen!");
	        Amount::ZERO
	    }    
        };
```